src/main/java/online/bookshop/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,137 @@
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import online.bookshop.dto.order.OrderRequestDto;
+import online.bookshop.dto.order.OrderResponseDto;
+import online.bookshop.dto.order.OrderUpdateStatusDto;
+import online.bookshop.dto.orderitem.OrderItemResponseDto;
+import online.bookshop.exeption.EntityNotFoundException;
+import online.bookshop.mapper.OrderItemMapper;
+import online.bookshop.mapper.OrderMapper;
+import online.bookshop.model.Order;
+import online.bookshop.model.OrderItem;
+import online.bookshop.model.ShoppingCart;
+import online.bookshop.repository.OrderItemRepository;
+import online.bookshop.repository.OrderRepository;
+import online.bookshop.repository.ShoppingCartRepository;
+import online.bookshop.service.OrderService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    @Transactional
+    public OrderResponseDto create(Long id, OrderRequestDto requestDto) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(id)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can't find shopping cart by userID: " + id));
+        Order order = new Order();
+        order.setUser(shoppingCart.getUser());
+        order.setStatus(Order.Status.NEW_ORDER);
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(requestDto.getShippingAddress());
+        Set<OrderItem> orderItems =
+                getOrderItemsFromShoppingCart(shoppingCart, order);
+        order.setOrderItems(orderItems);
+        BigDecimal total = calculateTotal(orderItems);
+        order.setTotal(total);
+        orderRepository.save(order);
+        orderItemRepository.saveAll(orderItems);
+        OrderResponseDto orderResponseDto
+                = orderMapper.toDto(order);
+        orderResponseDto.setOrderItems(changeToItemResponseDto(orderItems));
+        return orderResponseDto;
+    }
+
+    @Override
+    @Transactional
+    public List<OrderResponseDto> findAllByUserId(Long userId, Pageable pageable) {
+        List<Order> allByUserId = orderRepository.findAllByUserId(userId);
+        return allByUserId.stream()
+                .map(order -> {
+                    OrderResponseDto orderResponseDto = orderMapper.toDto(order);
+                    Set<OrderItemResponseDto> orderItemResponseDtos
+                            = changeToItemResponseDto(order.getOrderItems());
+                    orderResponseDto.setOrderItems(orderItemResponseDtos);
+                    return orderResponseDto;
+                })
```

Comment:
make a mapper for this

Reply:
i do in changeToItemResponseDto

Reply:
mapper should do it 
![image](https://github.com/OlexiiMelnik/book-shop/assets/20326858/87345a00-5ee3-45dc-86b0-b05734bfe3f6)